sqlScript.sql

```diff
@@ -24,17 +25,42 @@ ADD CONSTRAINT minDescLength CHECK (LENGTH(description) > 0);
 ALTER TABLE Course
 ADD CONSTRAINT minInfoLength CHECK (LENGTH(title) > 0);
 
+ALTER TABLE Course
+ADD CONSTRAINT minMaxAtt CHECK (maxAtt > 0);
+
+delimiter //
+
+CREATE TRIGGER trig_date_check BEFORE INSERT ON Course
+FOR EACH ROW
+BEGIN
+	IF (NEW.date < CURRENT_DATE()) = 1 THEN
+		SET NEW.title = null;
+	END IF;
+END //
+
+delimiter ;
+
 CREATE TABLE User(
 	userID int unsigned PRIMARY KEY auto_increment,
 	name varchar(30) not null,
-	email varchar(254) not null
+	email varchar(254) not null unique
 );
 
 ALTER TABLE User
 ADD CONSTRAINT minNameLength CHECK (LENGTH(name) > 0);
 
+delimiter //
+
+CREATE TRIGGER trig_email_check BEFORE INSERT ON User
+FOR EACH ROW
+BEGIN
+	IF (NEW.email REGEXP '^[^@]*[@]{1}[k]{1}[a]{1}[i]{1}[n]{1}[o]{1}[s]{1}[.]{1}[c]{1}[o]{1}[m]{1}$') = 0 THEN
```

Comment:
Great to see regex here! Only problem is this will allow `@kainos.com` which isn't a valid email address. If you change * to + it'll enforce the need for at least one character before `@`.